### PR TITLE
Optimize performance for some basic APIs

### PR DESCRIPTION
**Pull request by wxbty**
## What is the purpose of the change
Simply increase the cache for immutable beans or properties .
Can improve the performance of these frequently used APIs

### CODE REVIEW

File: dubbo-common/src/main/java/org/apache/dubbo/common/beans/factory/ScopeBeanFactory.java [hunk 1]
```diff
@@ -53,7 +54,9 @@ public class ScopeBeanFactory {
     private final List<BeanInfo> registeredBeanInfos = new CopyOnWriteArrayList<>();
     private InstantiationStrategy instantiationStrategy;
     private final AtomicBoolean destroyed = new AtomicBoolean();
-    private List<Class<?>> registeredClasses = new ArrayList<>();
+    private final List<Class<?>> registeredClasses = new ArrayList<>();
+
+    private final Map<Class<?>, Object> registeredCacheBeanMap = new ConcurrentHashMap<>(64);
```
Comment (AlbumenJ) [inline dubbo-common/src/main/java/org/apache/dubbo/common/beans/factory/ScopeBeanFactory.java:59]: Cache here should consider different bean with same class but different name

File: dubbo-common/src/main/java/org/apache/dubbo/rpc/model/ApplicationModel.java [hunk 2]
```diff
@@ -234,12 +236,18 @@ public ApplicationConfig getCurrentConfig() {
     }
 
     public String getApplicationName() {
-        return getCurrentConfig().getName();
+        if (StringUtils.isBlank(this.name)) {
+            this.name = getCurrentConfig().getName();
+        }
+        return this.name;
```
Comment (AlbumenJ) [inline dubbo-common/src/main/java/org/apache/dubbo/rpc/model/ApplicationModel.java:242]: ApplicationName and ApplicationModel's name are different

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-common/src/main/java/org/apache/dubbo/common/beans/factory/ScopeBeanFactory.java (hunk 1 above)
Comment (wxbty) [reply dubbo-common/src/main/java/org/apache/dubbo/common/beans/factory/ScopeBeanFactory.java:59]: Only beans without names are cached here. If most beans have no names, is this enough? Maybe changing the name of the map to NoNameBeanMap is more appropriate

File: dubbo-common/src/main/java/org/apache/dubbo/rpc/model/ApplicationModel.java (hunk 2 above)
Comment (wxbty) [reply dubbo-common/src/main/java/org/apache/dubbo/rpc/model/ApplicationModel.java]: ok

File: dubbo-common/src/main/java/org/apache/dubbo/common/beans/factory/ScopeBeanFactory.java (hunk 1 above)
Comment (wxbty) [reply dubbo-common/src/main/java/org/apache/dubbo/common/beans/factory/ScopeBeanFactory.java]: Aactually is the only unique Name

File: dubbo-common/src/main/java/org/apache/dubbo/rpc/model/ApplicationModel.java [hunk 3]
```diff
@@ -234,12 +236,18 @@ public ApplicationConfig getCurrentConfig() {
     }
 
     public String getApplicationName() {
-        return getCurrentConfig().getName();
+        if (StringUtils.isBlank(this.applicationName)) {
+            this.applicationName = getCurrentConfig().getName();
+        }
+        return this.applicationName;
```
Comment (AlbumenJ) [inline dubbo-common/src/main/java/org/apache/dubbo/rpc/model/ApplicationModel.java:242]: `ApplicationName` in config manager may changed in runtime.

File: dubbo-common/src/main/java/org/apache/dubbo/common/beans/factory/ScopeBeanFactory.java [hunk 4]
```diff
@@ -197,8 +200,16 @@ public <T> List<T> getBeansOfType(Class<T> type) {
...
     public <T> T getBean(Class<T> type) {
-        return this.getBean(null, type);
+        if (registeredCacheBeansWithUniqueName.containsKey(type)) {
+            return (T) registeredCacheBeansWithUniqueName.get(type);
+        }
+        T bean = this.getBean(null, type);
+        if (bean != null) {
+            registeredCacheBeansWithUniqueName.put(type, bean);
+        }
+        return bean;
```
Comment (AlbumenJ) [inline dubbo-common/src/main/java/org/apache/dubbo/common/beans/factory/ScopeBeanFactory.java:212]: Is there a better way to improve the performance in `BeanFactory`

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

**Pull request closed without merging**